### DISCUSSION TITLE
`show_labels` and `show_conf` don't work for the `mode.train` output

### DISCUSSION THREAD

**Discussion by tonyreina** [ultralytics/ultralytics#18401] [Q&A]
I'm using the standard `model.train` and I specifically pass in the flags `show_labels=False` and `show_conf=False` but in the training output directory it still adds labels to the images. Is there another way to suppress the labels?

```
# Train the model
    results = model.train(
        data=yaml_file,
        model=model_base,
        epochs=epochs,
        batch=batch_size,
        imgsz=640,
        name=model_name,
        project=project_name,
        single_cls=True,
        mosaic=0.2,
        translate=0.1,
        flipud=0.5,
        fliplr=0.5,
        scale=0.1,
        seed=816,
        resume=False,
        patience=patience,
        tracker="bytetrack",
        half=True,
        line_width=1,
        show_labels=False,
        show_conf=False, 
    )
    ```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tonyreina, thank you for bringing this up and for using Ultralytics 🚀! We appreciate your detailed query. It looks like you're encountering unexpected behavior with the `show_labels` and `show_conf` flags during training output visualization.

If this is a 🐛 Bug Report:
- Could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) that replicates this behavior? This will help our team debug more effectively. Include the exact command or script snippet you're running along with any relevant environment details.

Meanwhile, here are a few steps you can try to troubleshoot and ensure you're using the latest version of Ultralytics:
### Upgrade
Make sure you're running the latest release:
```bash
pip install -U ultralytics
```
This ensures any recent bug fixes or updates are included in your environment.

### Documentation
Double-check the expected behavior for `show_labels` and `show_conf` in the [Docs](https://docs.ultralytics.com). You may also explore our [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples for additional guidance.

### Environments
To verify reproducibility, you can also test the workflow across one of our verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image**: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a> See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)

### Community Support
Feel free to join the Ultralytics community to discuss this further:
- [Discord](https://discord.com/invite/ultralytics) for real-time assistance 🎧
- [Discourse](https://community.ultralytics.com) for detailed discussions and explorations
- [Subreddit](https://reddit.com/r/Ultralytics) to share insights with fellow YOLO enthusiasts

### Status
Lastly, make sure our CI tests are green:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>
This ensures that YOLO's [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are working correctly in all major environments.

This is an automated response to assist you quickly, and an Ultralytics engineer will follow up with you soon to provide further support 😊. Stay tuned!

**Comment by tonyreina** [comment]


**Reply by glenn-jocher** [reply]
@tonyreina it seems the image upload did not complete. Could you re-upload or provide additional context so we can assist you better?

**Comment by Y-T-G** [comment]
`show_labels` and `show_conf` only applies to `model.predict()`. You can't prevent the labels and conf in training output.
